sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -633,6 +651,15 @@ kernel_pid_t gcoap_init(void)
 
 void gcoap_register_listener(gcoap_listener_t *listener)
 {
+#ifdef MODULE_GCOAP_LAZY_INIT
```

Comment:
If a node is not registering additional resources (I know, maybe not the most common of cases), and acts only as a client, then it won't be able to perform CON requests right?

Reply:
Yup. This is checked in `gcoap_req_send()`, if a CON request is made, it will fail to send:

https://github.com/RIOT-OS/RIOT/blob/49d1b9e0880b33ed9c8995df877ee472564b7a2d/sys/net/application_layer/gcoap/gcoap.c#L760-L765

Reply:
Yes, I saw that. Is that a wanted behaviour?: Every node must register at least one listener to be able to send CON requests.

Reply:
Maybe we could find another way to trigger the server to start

Reply:
Alternatively, instead of returning an error, the server can be started from `gcoap_req_send()` when the client wants to send a CON request. The main goal of lazy init is the delayed start of the server until it is really needed. So starting the server from `gcoap_req_send()` still fits in the requirement.

Reply:
O.k. @leandrolanzieri ?